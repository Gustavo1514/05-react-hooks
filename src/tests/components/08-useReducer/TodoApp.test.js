const { act } = require('@testing-library/react')
const { shallow, mount } = require('enzyme')
const { default: TodoApp } = require('../../../components/08-useReducer/TodoApp')

const { demoTodos } = require('../../fixtures/demoTodos')
describe('pruebas en <TodoApp/>', () => {

  const wrapper = shallow(<TodoApp />)
  Storage.prototype.setItem= jest.fn(()=>{})
  test('debe mostrarse correctamente', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('debe agregar un Todo', () => {
    const wrapper = mount(<TodoApp />)
    act(()=>{
      wrapper.find('TodoAdd').prop('handleAddTodo')(demoTodos[0])
      wrapper.find('TodoAdd').prop('handleAddTodo')(demoTodos[1])

    })
    expect(wrapper.find('h1').text().trim()).toBe('Todo App (2)')
    expect(localStorage.setItem).toHaveBeenCalledTimes(2)
  })

  test('debe eliminar un todo', () => {
    wrapper.find('TodoAdd').prop('handleAddTodo')(demoTodos[0])
    wrapper.find('TodoList').prop('handleDelete')(demoTodos[0].id)
    expect(wrapper.find('h1').text().trim()).toBe('Todo App (0)')

  });
})
