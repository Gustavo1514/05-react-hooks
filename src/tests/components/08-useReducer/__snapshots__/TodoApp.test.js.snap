// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <TodoApp/> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Todo App 
    <small>
      (
      0
      )
    </small>
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col-sm-7"
    >
      <TodoList
        handleDelete={[Function]}
        handleToogle={[Function]}
        todos={Array []}
      />
    </div>
    <div
      className="col-sm-5"
    >
      <TodoAdd
        handleAddTodo={[Function]}
      />
    </div>
  </div>
</div>
`;
